api/agent/agent.go

```diff
@@ -417,14 +416,14 @@ func (a *agent) waitHot(ctx context.Context, call *call) (Slot, error) {
 // launchCold waits for necessary resources to launch a new container, then
 // returns the slot for that new container to run the request on.
 func (a *agent) launchCold(ctx context.Context, call *call) (Slot, error) {
-
 	isAsync := call.Type == models.TypeAsync
 	ch := make(chan Slot)
-	ctxResource, cancelResource := context.WithCancel(ctx)
-	defer cancelResource()
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "agent_launch_cold")
+	defer span.Finish()
 
 	select {
-	case tok, isOpen := <-a.resources.GetResourceToken(ctxResource, call.Memory, uint64(call.CPUs), isAsync):
+	case tok, isOpen := <-a.resources.GetResourceToken(ctx, call.Memory, uint64(call.CPUs), isAsync):
```

Comment:
side effect of resource cancel go routine to live longer. (before: go routine lived while waiting for resource, after: it waits call ctx duration)

Reply:
I believe this is also fixed here: https://github.com/fnproject/fn/commit/e258e3ebc28e3c3bb1f27d8e33897cab8f62b477 -- either we get a resource (and no goroutines live on) or we hit ctx.Done() in this scope (also shuts them down)